### PR TITLE
Ensure WARN as a warning log level

### LINKED ISSUE #3: Ensure WARN as a warning log level
It was already fixed in:
- eclipse-kanto/file-upload#45
- eclipse-kanto/software-update#16

It has to be ensured that WARN is used as a warning log level.

### PR DESCRIPTION
**Pull request by velitchko-valkov**
[#3] Ensure WARN as a warning log level
 - Changed dependency to latest File Upload version with warn log change
 - Adapted module to use 'warn' and 'warnf' function names

Signed-off-by: Velitchko Valkov <Velitchko.Valkov@bosch.io>

### CODE REVIEW

**Comment by e-grigorov** [comment]
@velitchko-valkov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).

This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances :blush: 

Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

File: go.sum [hunk 1]
```diff
@@ -52,14 +52,16 @@ github.com/davecgh/go-spew v1.1.1/go.mod h1:J7Y8YcW2NihsgmVo/mv3lAwl/skON4iLHjSs
...
 github.com/dnaeon/go-vcr v1.2.0/go.mod h1:R4UdLID7HZT3taECzJs4YgbbH6PIGXB6W/sc5OLb6RQ=
-github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220803092511-39328590c3c8 h1:2+5krbaKzvk9zSI3Bggti7USDbnjskoXyf05BYna7eA=
-github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220803092511-39328590c3c8/go.mod h1:Rhqfpu156wHPP+6NQMzkfWFhJYR3SeIiQaRIi0c2zg0=
+github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220819091951-08d3c358eeb6 h1:BcL6XT/UnteQJLtF34aOG/2PZBqG44uHwGrD6SG0uwo=
+github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220819091951-08d3c358eeb6/go.mod h1:9JAFvJxUIFgFAm+gtp+BYE1s/NDo9diQp+3y3F+9eHQ=
 github.com/eclipse/ditto-clients-golang v0.0.0-20211126080253-3e1c6149fede/go.mod h1:hAXWyOdJLxUQTK4nCc3xAHgEU5iQa5a5DzFeS8ErR60=
 github.com/eclipse/ditto-clients-golang v0.0.0-20211126080925-0676267c80ac h1:+Tl05cZOU8RXl95RP2iZ6LFM5CXxTVhz4ux0LtPUYR4=
 github.com/eclipse/ditto-clients-golang v0.0.0-20211126080925-0676267c80ac/go.mod h1:hAXWyOdJLxUQTK4nCc3xAHgEU5iQa5a5DzFeS8ErR60=
 github.com/eclipse/paho.mqtt.golang v1.2.0/go.mod h1:H9keYFcgq3Qr5OUJm/JZI/i6U7joQ8SYLhZwfeOo6Ts=
 github.com/eclipse/paho.mqtt.golang v1.3.5 h1:sWtmgNxYM9P2sP+xEItMozsR3w0cqZFlqnNN1bdl41Y=
```
Comment (e-grigorov) [inline go.sum]: Could you please run, go mod tidy? It should remove 1.3.5 references.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Comment by velitchko-valkov** [comment]
> @velitchko-valkov The Eclipse Kanto project is currently restructuring its license scheme - from Eclipse Public License 2.0 only to dual licensing under Eclipse Public License 2.0 **and** Apache License 2.0 (you can read more about it [here](https://projects.eclipse.org/projects/iot.kanto/reviews/restructuring-review)).
> 
> This contribution will be merged into the project with this new licensing scheme so its important for the project and the community to have your agreement on proceeding with it given these new circumstances 😊
> 
> Please, provide your feedback as a comment here that you, as an author of this contribution, understand and agree with the change explained above. Thank you!

I understand and agree with the license change explained above.

File: go.sum [hunk 2]
```diff
@@ -102,6 +102,7 @@ golang.org/x/net v0.0.0-20210805182204-aaa1db679c0d/go.mod h1:9nx3DQGgdP8bBQD5qx
 golang.org/x/net v0.0.0-20211123203042-d83791d6bcd9 h1:0qxwC5n+ttVOINCBeRHO0nq9X7uy8SDsPoi5OaCdIEI=
 golang.org/x/net v0.0.0-20211123203042-d83791d6bcd9/go.mod h1:9nx3DQGgdP8bBQD5qxJ1jj9UTztislL4KSBs9R2vV5Y=
 golang.org/x/sync v0.0.0-20190423024810-112230192c58/go.mod h1:RxMgew5VJxzue5/jJTE5uejpjVlOe/izrB70Jof72aM=
+golang.org/x/sync v0.0.0-20210220032951-036812b2e83c h1:5KslGYwFpkhGh+Q16bwMP3cOontH8FOep7tGV86Y7SQ=
```
Comment (e-grigorov) [inline go.sum:105]: If it's a runtime dependency, we have to add it to the notice file.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

File: go.sum (hunk 2 above)
Comment (velitchko-valkov) [reply go.sum:105]: I'll check.

File: go.mod [hunk 3]
```diff
@@ -3,9 +3,9 @@ module github.com/eclipse-kanto/file-backup
 go 1.17
 
 require (
-	github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220803092511-39328590c3c8
+	github.com/eclipse-kanto/file-upload v0.1.0-M1.0.20220819091951-08d3c358eeb6
 	github.com/eclipse/ditto-clients-golang v0.0.0-20211126080925-0676267c80ac
-	github.com/eclipse/paho.mqtt.golang v1.3.5
+	github.com/eclipse/paho.mqtt.golang v1.4.1
```
Comment (e-grigorov) [inline go.mod:8]: Eclipse Paho has to be updated to 1.4.1 in the notice file.

**Review by e-grigorov — CHANGES_REQUESTED** [review]

**Review by e-grigorov — APPROVED** [review]

**Pull request merged**